src/task/screen-bad-words-task.ts

```diff
@@ -0,0 +1,51 @@
+// global
+import { DatabaseTransactionHandler, Member } from 'graasp';
+import BadWordsFilter from 'bad-words';
+import { stripHtml } from '../utils';
+// local
+import { ValidationService } from '../db-service';
+import { BaseValidationTask } from '../task/base-validation-task';
+
+type InputType = { itemId: string };
+
+export class ScreenBadWordsTask extends BaseValidationTask<boolean> {
+  input: InputType;
+
+  get name(): string {
+    return ScreenBadWordsTask.name;
+  }
+
+  constructor(member: Member, validationService: ValidationService, input: InputType) {
+    super(member, validationService);
+    this.input = input;
+  }
+
+  checkBadWrods = (documents: string[]) => {
+    const strings = documents?.filter(Boolean).map((entry) => stripHtml(entry));
+    const badWordsFilter = new BadWordsFilter();
+    for (const index in strings) {
+      if (badWordsFilter.isProfane(strings[index])) {
+        return true;
+      }
+    }
+    return false;
+  };
+
+  async run(handler: DatabaseTransactionHandler): Promise<void> {
+    this.status = 'RUNNING';
+
+    const { itemId } = this.input;
+    const item = await this.validationService.getItem(itemId, handler);
```

Comment:
Here you could use itemService (that you pass from the plugin to the constructor) `this.itemService.getItem(...)`